azure/scope/machine.go

```diff
@@ -352,6 +353,7 @@ func (m *MachineScope) RoleAssignmentSpecs(principalID *string) []azure.Resource
 			Scope:            m.SystemAssignedIdentityScope(),
 			RoleDefinitionID: m.SystemAssignedIdentityDefinitionID(),
 			PrincipalID:      principalID,
+			PrincipalType:    armauthorization.PrincipalTypeServicePrincipal,
```

Comment:
Instead of hardcoding `PrincipalType:    armauthorization.PrincipalTypeServicePrincipal`, would it be valuable to have the user specify `PrincipalType` in the MachinePool spec?

Is it possible that `PrincipleType` can take other values than `PrincipalTypeServicePrincipal` ?

Reply:
That's a great question. Thanks for reviewing @nawazkh!

I found these possible values for [PrincipalType](https://pkg.go.dev/github.com/Azure/azure-sdk-for-go/sdk/resourcemanager/authorization/armauthorization/v2@v2.2.0#PrincipalType) 

```go
const (
	PrincipalTypeDevice            = "Device"
	PrincipalTypeForeignGroup      = "ForeignGroup"
	PrincipalTypeGroup             = "Group"
	PrincipalTypeServicePrincipal  = "ServicePrincipal"
	PrincipalTypeUser              = "User"
)
```
and it seemed that the SP type would be associated with the `PrincipalID`, but it's not entirely clear to me. @whites11 can you clarify?

Reply:
Sorry for the delay, I missed the notification.

So, this role assignment is used only for system assigned role for VMs/VMSSes.
In this scenario, as far as I understood and as far as my tests proved, the only meaningful value for the principal type is `service principal`.

Reply:
Thanks for the explanation. I agree: none of the other values seemed relevant, and this is limited in scope.